src/Utils/ProductGalleryUtils.php

```diff
@@ -129,17 +129,34 @@ public static function maybe_generate_intermediate_image( $attachment_id, $size
 		$image_path     = wp_get_original_image_path( $attachment_id );
 		$image_metadata = wp_get_attachment_metadata( $attachment_id );
 
-		// If image doesn't exist, we can't generate the intermediate size. Bail.
-		if ( ! isset( $image_metadata['path'] ) ) {
+		// If image sizes are not available. Bail.
+		if ( ! isset( $image_metadata['width'], $image_metadata['height'] ) ) {
 			return;
 		}
 
 		/*
 		 * We want to take the minimum dimension of the image and
 		 * use that size as the crop size for the new image.
 		 */
-		$min_size                         = min( $image_metadata['width'], $image_metadata['height'] );
-		$new_image_metadata               = image_make_intermediate_size( $image_path, $min_size, $min_size, true );
+		$min_size = min( $image_metadata['width'], $image_metadata['height'] );
+
+		/*
+		 * By default WordPress does not create a new intermediate image
+		 * if the ratio is off by 1 pixel i.e. 500x501 image.
+		 * We need to allow discrepancies of 1 pixel to proceed with the cropping
+		 * so it stays consistent.
+		 */
+		add_filter( 'wp_image_resize_identical_dimensions', '__return_true' );
```

Comment:
According to the documentation, this filter:

> Filters whether to proceed with making an image sub-size with identical dimensions with the original/source image. Differences of 1px may be due to rounding and are ignored.

It appears that if we always return true, it will create new subsizes even for images that are already squared or when the original size matches the size we want to create.

While testing this card, I noticed that squared images are being regenerated even though they shouldn't be. I believe it is necessary to enhance the condition to return `false` if:

- The original image size is identical to the target image size or the image is already an squared image. This is to prevent situations where we attempt to create a resized image of 500x500 when the original image already has a size of 500x500.

Reply:
Thanks for reviewing. We've decided not to make any changes so I will close this PR.

Reply:
> This is to prevent situations where we attempt to create a resized image of 500x500 when the original image already has a size of 500x500.

FYI - WP function won't create another one if there is already one with the same size.